types/types.go

```diff
@@ -186,6 +186,10 @@ func IsValidAddress(address string) error {
 		return fmt.Errorf("address %s has invalid length", string(decodedAddress))
 	}
 
+	if addr := StringToAddress(address); addr == ZeroAddress && !zeroAddressAllowed {
```

Comment:
I'd suggest returning the parsed address since we are parsing it in case it is valid.

Reply:
Fixed https://github.com/Ethernal-Tech/blade/pull/56/commits/f21a17a9c3b56e71fc50ef0dcf3665cfb67a6b78